keps/sig-apps/3088-pdb-for-daemonset-pods/README.md

```diff
@@ -0,0 +1,265 @@
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+Support podDusruptionBudget (PDB) for DaemonSet pods. DaemonSet pods are evicted all at once
+ignoring PDB. It creates a load spike for kube-apiserver especially in relatively big clusters. This
+KEP proposes to change this behavior so that users could rely on PDB for DaemonSet pods like for all
+other built-in controllers.
+
+## Motivation
+
+PodDisruptionBudget can be not only a tool to keep user apps available during evictions, but also a
+mean of reducing of load on kube-apiserver. The can be caused by a restart of big number of
+DaemonSet pods due to eviction.
+
+Additionally, the disruption controller tracks all built-in pod controllers except DaemonSets when
+calculates allowed disruption. This behavior in inconsistent across built-in pod controllers and may
+cause confusion.
+
+### Goals
+
+- Support the disruption budget for DaemonSet the same way it works for other built-in pod
```

Comment:
If we have any evidence that assumptions have been made in daemonset that prevent arbitrary controllers working with pods and PDB, that's a bug.  If we wish to enhance the behavior of daemonsets, that's desirable.